### DISCUSSION TITLE
Could there be an option to fallback to the [artist] tag (rather than "Various Artists") if [albumartist] is not set?

### DISCUSSION THREAD

**Discussion by SteveDinn** [music-assistant/support#1109] [Feature requests and ideas]
My home assistant log is littered with entries like these:
```
2023-02-28 10:23:42.906 WARNING (MainThread) [music_assistant.music.file] <path-to-file-here> is missing ID3 tag [albumartist], using Various Artists as fallback
```

My albums, if they are actually comprised of various artists, will have the [albumartist] tag explicitly set to "Various Artists". Whereas a large number of my single-artist albums, will only have the [artist] tag and not [albumartist]. I get the the need to have some fallback, but could we get an option to fall back to the [artist] tag instead?

**Comment by OzGav** [comment]
What we actually recommend is that all music is tagged comprehensively. See the section here on Tagging Local Music https://github.com/music-assistant/hass-music-assistant/discussions/820

**Reply by SteveDinn** [reply]
While I agree with this in principle, I don't know of a way to make this change to my entire library. Specifically, I would want to set the [albumartist] to the [artist] if [albumartist] was not previously set for my entire library of FLAC and MP3 files. Does there exist a script or tool to make this change en masse?

**Reply by SteveDinn** [reply]
I don't just want to re-tag everything because I have painstakingly updated many, many tags manually, so I want something to make that one specific change.

**Reply by OzGav** [reply]
I understand. I did go through the process of tagging all my music with MusicBrainz Picard and it does take a while if you have a lot of albums. However, you will not be able to get the full benefit of MA without doing so. MA uses tags that contain unique identifiers for the artists/albums/tracks that allows linking across platforms. Tags such as ISRC and MusicBrainz IDs.
I am not aware of a script that will selectively update a tag but you might be lucky that someone on the Internet has done such a thing.
But again we strongly recommend you tag all your music comprehensively with software that pulls data from a centralised database. This ensures you don't have issues such as mis- or alternate spellings for example.

**Reply by SteveDinn** [reply]
Totally with you. All my newer music goes through Beets. I can live with the problem for my older stuff.

Thanks again.

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
